src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1288,14 +1285,21 @@ private void CalculateIORedirection(out bool redirectOutput, out bool redirectEr
             }
         }
 
-        private bool ValidateExtension(string path)
+        // On Windows, check the extension list and see if we should try to execute this directly.
+        // Otherwise, use the platform library to check executability
+        private bool IsExecutable(string path)
         {
-            // Now check the extension and see if it's one of the ones in pathext
+#if UNIX
+
+            if (Platform.NonWindowsIsExecutable(this.Path))
+            {
+                return true;
+            }
+#else
+
             string myExtension = System.IO.Path.GetExtension(path);
 
-            string pathext = (string)LanguagePrimitives.ConvertTo(
-                this.Command.Context.GetVariableValue(SpecialVariables.PathExtVarPath),
-                typeof(string), CultureInfo.InvariantCulture);
+            var pathext = Environment.GetEnvironmentVariable("PATHEXT");
```

Comment:
~~It might be messier, but figured I'd ask.~~

~~We have a constant for this -- we could keep that defined on *nix and use it here instead of the string literal. It just might be confusing in SpecialVariables.cs that we have the constant and no variable. But that is what comments are for!~~

Reading through it again I see the constant variable name begins with `env:`. Seems better to me to not mix this line with that file.

Reply:
Sorry, should I fix anything?

Reply:
No -- I think you've done it the best way!